doc/developer/design/README.md

```diff
@@ -40,20 +32,33 @@ Examples:
 * [Cluster SQL API](https://github.com/MaterializeInc/materialize/pull/10680)
 * [`WITH MUTUALLY RECURSIVE`](https://github.com/MaterializeInc/materialize/pull/16445)
 
+## Goals
+
+Summary of the goals of a design document:
+
+ - Record context and assumptions.
+ - Achieve a (reasonable) level of consensus around a design.
+ - Record decisions.
+
 ## When should you make a design document?
 
 Design docs should be written for all changes where an implementation does not
 immediately follow from the problem.
 
 Some specific times when you should write a design document:
 
+Definitely write a design document:
+
+1. If it's going to involve multiple people and/or teams coordinating changes.
```

Comment:
Do we want to mandate a design doc already when there is just two people working on a feature together?

Reply:
should probably move that to "consider writing a design doc". You're right